clients/source.go

```diff
@@ -211,9 +216,16 @@ func (c *SourceClient) Terminate() error {
 	}
 	if c.cmd != nil && c.cmd.Process != nil {
 		if err := c.cmd.Process.Kill(); err != nil {
+			// if we fail to kill the process, we also won't wait for logs to finish streaming
```

Comment:
I don't think this comment is correct or at least we need to see what are the error message from `kill`.

This is the snippet from signal, it really can only fail if it already closed. you can always kill a child process. so we do want to wait for the logs if it is already killed.
```
func (p *Process) signal(sig Signal) error {
	if p.Pid == -1 {
		return errors.New("os: process already released")
	}
	if p.Pid == 0 {
		return errors.New("os: process not initialized")
	}
	p.sigMu.RLock()
	defer p.sigMu.RUnlock()
	if p.done() {
		return ErrProcessDone
	}
	s, ok := sig.(syscall.Signal)
	if !ok {
		return errors.New("os: unsupported signal type")
	}
	if e := syscall.Kill(p.Pid, s); e != nil {
		if e == syscall.ESRCH {
			return ErrProcessDone
		}
		return e
	}
	return nil
}
```

Reply:
@yevgenypats Updated